adr/adr-responses-api.md

```diff
@@ -0,0 +1,45 @@
+
+The core architectural question is: How can we build a robust, developer-friendly agentic framework on an API that does not provide an explicit `Agent` construct without sacrificing the usability that such a construct provides?
+
+One issue that is *not* in scope for this document and is not affected by this decision is whether tools will be invoked by the server or by the client.  For both APIs, it is the case that:
+
+* MCP tools are invoked by the server.  The Responses API can be configured to get approval from the client before invoking the tool, but with either configuration, the tool is invoked by the server.
+* "Hosted" or "built-in" tools are also invoked by the server.
+* "Function" tools are invoked by the client.
+
+So security and network access concerns for tool calling are outside of the scope of this document because they are the same concerns regardless of which API is used.
+
+## Decision
+
+We will adopt the OpenAI-compatible **Llama Stack Responses API** as the exclusive server interface for agentic reasoning within Red Hat's AI products.
+
+To address the absence of a server-side agent construct, we will need **a client-side Agent construct**.  To address the absence of a server-side agent construct, we will need **a client-side Agent construct**.  The decision about its form is addressed in [ADR: Adopt a Minimal SDK and Ecosystem Strategy for Agents](https://github.com/opendatahub-io/agents/pull/2). Some proposals include adding the functionality to the official Llama Stack client library, creating a new SDK maintained by Red Hat (perhaps as part of OpenDataHub), providing a container image with a simple HTTP API, and adopting a pre-existing SDK such as LangChain. That ADR will cover the pros and cons of these options and propose a final resolution.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Strategic Alignment & Future-Proofing:** By building on the OpenAI-compatible Responses API, we align with the industry standard and the strategic direction of Llama Stack. This de-risks our platform from the likely deprecation of the older Agents API.
+* **Ecosystem Compatibility:** Our agentic capability will be natively compatible with the vast ecosystem of tools, libraries, and monitoring solutions that are built to interface with the OpenAI API format. This accelerates both internal development and customer adoption.
+* **Enhanced Flexibility and Control:** A client-side agent construct gives the application developer full control over the agent's execution loop and tool selection. This is critical for complex, mission-critical applications where logic cannot be a black box on a server.
```

Comment:
This isn't necessarily the case, it depends on the abstraction offered.

Reply:
Perhaps add a bit more detail here - 
"The Responses API offers useful abstractions to simplify client-side code with its default settings, but also allows for fine-grained control over those abstractions, which makes it flexible enough for more complex applications."

Reply:
I added this text from @amfred but also softened the wording of the earlier text because I agree with @grs that it was overstated.